### PR TITLE
[NVSHAS-9630] Remove redundant gha in scan-action

### PR DESCRIPTION
**Pull request by pohanhuangtw**
### Summary

- Add release.yaml to auto prepare test env, and release.

### CODE REVIEW

File: .github/workflows/build.yaml
Comment (holyspectral) [inline .github/workflows/build.yaml]: What will the bats version it will be? In my Ubuntu 22.04, it's `1.2.1-3build1`, which seems pretty old.

The original setup-bats gets the bats from https://github.com/bats-core/bats-core/archive/v${version}.tar.gz

File: .github/workflows/build.yaml
Comment (holyspectral) [inline .github/workflows/build.yaml:23]: Have you considered getting hadolint from their repo instead? https://github.com/hadolint/hadolint/releases/tag/v2.12.0

**Review by holyspectral — COMMENTED** [review]
Generally looks good to me.  A couple of questions about the version of dependencies.

File: .github/workflows/build.yaml
Comment (pohanhuangtw) [reply .github/workflows/build.yaml]: Hi @holyspectral ,
Thanks for the review, Let me find a way to retain the current version. As far as I know, our original setting doesn’t have a specific reason for this particular version.

File: .github/workflows/build.yaml
Comment (pohanhuangtw) [reply .github/workflows/build.yaml:23]: No, I didn't. I believe other projects in neuvector haven't applied such actions, so we should remove it.

If we actually want to use it, should I request approval from security team first?

File: .github/workflows/build.yaml
Comment (pohanhuangtw) [reply .github/workflows/build.yaml]: Thanks for the advice, please take a look

File: .github/workflows/build.yaml
Comment (holyspectral) [reply .github/workflows/build.yaml:23]: I saw your change now.  Thanks for doing it!
Actually the change this time focuses on unapproved GHA, as it has a lot more privileges than one executable.  Using hadolint is fine.

File: .github/workflows/build.yaml
Comment (holyspectral) [inline .github/workflows/build.yaml]:
```suggestion
          wget -O hadolint https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-Linux-x86_64
          echo "56de6d5e5ec427e17b74fa48d51271c7fc0d61244bf5c90e828aab8362d55010  hadolint" | sha256sum --check
```

Comment (pohanhuangtw) [reply .github/workflows/build.yaml]: Thanks

File: .github/workflows/build.yaml
Comment (holyspectral) [inline .github/workflows/build.yaml:15]: How about using actions/checkout instead of using `git clone`?
```
      - name: Checkout BATS
        uses: actions/checkout@v4
        with:
          repository: bats-core/bats-core
          ref: v1.11.0
```

**Review by holyspectral — APPROVED** [review]
LGTM in terms of functionality.  `actions/checkout` is good to have.

**Review by holyspectral — APPROVED** [review]

**Comment by pohanhuangtw** [comment]
Hi @holyspectral ,
I squash it to one commit, please take a look

**Review by holyspectral — APPROVED** [review]

**Pull request merged**
